### PR TITLE
fix Issue 19998 - [REG2.086.0] [e2ir] Assertion failure with -profile…

### PR DESCRIPTION
**Pull request by WalterBright**
…=gc and -dip1008

### CODE REVIEW

**Comment by WalterBright** [comment]
The bug isn't in stable, so this stays in master.

File: test/compilable/test20318.d [hunk 1]
```diff
@@ -5,3 +5,9 @@ void main()
 {
     throw new Exception("msg");
 }
+
+void test19998()
+{
+    throw new Exception("");
```
Comment (RazvanN7) [inline test/compilable/test20318.d]: I don't get how this addition is any different from the already existing test.

**Comment by RazvanN7** [comment]
I'm confused about this. This issue seems identical to https://issues.dlang.org/show_bug.cgi?id=20318 which was fixed by https://github.com/dlang/dmd/pull/10517 . Am I missing something?

**Comment by MoonlightSentinel** [comment]
The example compiles & runs fine with current master

**Comment by RazvanN7** [comment]
@WalterBright ?

File: test/compilable/test20318.d (hunk 1 above)
Comment (WalterBright) [reply test/compilable/test20318.d]: It seems I fixed a duplicate of the bug you fixed. I prefer the comment you made, and my fix was slightly more efficient, so I combined the two.

**Comment by NilsLankila** [comment]
to add the test case but so far this has no value anymore as the problem is fixed.

**Pull request closed without merging**